### PR TITLE
Send pipeline_step_used event

### PR DESCRIPTION
**Pull request by i386**
# Description

Has the following properties:
* `type` e.g. org.jenkinsci.plugins.workflow.steps.EchoStep
* `timesUsed` count of times executed in the pipeline
* `isDeclarative` used with a declarative Pipeline or not
* `runResult` the result of the job it was used with

See [JENKINS-48392](https://issues.jenkins-ci.org/browse/JENKINS-48392).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java [hunk 1]
```diff
@@ -89,6 +91,15 @@ public String getType() {
         return node.getType().name();
     }
 
+    @Override
+    public String getAnalyticsType() {
+        if (node.getNode() instanceof StepAtomNode) {
```
Comment (abayer) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java]: I’d include `StepStartNode` in a perfect world, but we aren’t tracking those yet here in the first place.

File: blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineStep.java [hunk 2]
```diff
@@ -57,6 +57,11 @@
     @Exported(name = TYPE)
     public abstract String getType();
 
+    /**
+     * @return the type of step used for analytics purposes
+     */
+    public abstract String getAnalyticsType();
```
[the remark this answers is not in the document]
Comment (i386) [reply blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineStep.java]: Fixed

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java (hunk 1 above)
Comment (i386) [reply blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java]: @abayer I've updated this method to take into account `StepStartNode` as you have suggested.

File: blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineStep.java [hunk 3]
```diff
@@ -57,6 +57,11 @@
     @Exported(name = TYPE)
     public abstract String getType();
 
+    /**
+     * @return the type of step used for analytics purposes
```
Comment (vivek) [inline blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineStep.java]: Remove ‘used for analytics purposes’, doesn’t need to be here.

**Comment by i386** [comment]
Replaced by https://github.com/jenkinsci/blueocean-plugin/pull/1579/

**Pull request closed without merging**